### PR TITLE
feat: Fix getProcessOverviewsByKey query #23793

### LINKED ISSUE #23793: Process owner is not saved
Engineering DRI: @oleksandr-kriuchenko-lohika

### Describe the bug

When trying to add process owner to any process on Dashboards page it is possible to successfully use the process configuration modal to select a user and hit **Svae**, but this change is not being persisted.

[Screencast from 2024-10-22 08-40-53.webm](https://github.com/user-attachments/assets/aab2077b-5e4a-40a6-8399-975bc0370752)

[Screencast from 2024-10-23 14-51-34.webm](https://github.com/user-attachments/assets/107d9ca6-598f-4f7b-8054-c8a72dae048a)

Also see the [comment below](https://github.com/camunda/camunda/issues/23793#issuecomment-2428481568)

### To Reproduce

1. Open Dashboards page
2. In the processes list, for any process click on the gear icon to open a config modal
3. in the modal select a process owner and hit save
4. this change is not displayed neither straight away nor when refreshed

### Environment observed:

- Optimize mode(s): C8, C7
- Optimize version(s): latest (but needs checking)
- Database(s) (Elasticsearch/OpenSearch): Elasticsearch/Opensearch

### Breakdown

```[tasklist]
### Pull Requests
- [ ] https://github.com/camunda/camunda/pull/24189
- [ ] https://github.com/camunda/camunda/pull/24219
- [ ] https://github.com/camunda/camunda-optimize/pull/14601#issuecomment-2447765133
- [ ] https://github.com/camunda/camunda-optimize/pull/14603
```

## Bug Lifecycle

For managing the issue lifecycle, please use the workflow commands. You can see the available
commands by writing `/help` as a comment on this issue.

### QA Verification

Is this bug reproducible?

- [ ] Yes. If so:
  - [ ] The bug description is clear
  - [ ] The steps to reproduce are clear
  - [ ] The environments observed are correct and complete
- [ ] No

#### Engineering Review

- [ ] All code targeting the main branch has been reviewed by at least one Engineer
- [ ] The PR targeting the main branch has been approved by the reviewing Engineer
- [ ] If the API has changed, the API documentation is updated
- [ ] All other PRs (docs, controller etc.) in the breakdown have been approved

#### QA Review

- [ ] The change is implemented as described on all target environments/versions/modes
- [ ] The documentation changes are as expected

### Completion

- [ ] All Review stages are successfully completed
- [ ] All associated PRs are merged to the main branch(es) and stable branches
- [ ] The correct version labels are applied to the issue

### PR DESCRIPTION
**Pull request by oleksandr-kriuchenko-lohika**
## Description

Fix getProcessOverviewsByKey query

## Checklist

- [ ] for CI changes:
  - [ ] structural/foundational changes signed off by [CI DRI](https://github.com/cmur2)
  - [ ] [ci.yml](https://github.com/camunda/camunda/blob/main/.github/workflows/ci.yml) modifications comply with ["Unified CI" requirements](https://github.com/camunda/camunda/wiki/CI-&-Automation#workflow-inclusion-criteria)

## Related issues

closes #23793

### CODE REVIEW

**Review by grlimacan — APPROVED** [review]
LGTM, this should also be backported to stable/optimize-8.6 though . Is this error also present in C7?

File: optimize/backend/src/main/java/io/camunda/optimize/service/db/repository/es/ProcessRepositoryES.java [hunk 1]
```diff
@@ -59,7 +59,8 @@ public Map<String, ProcessOverviewDto> getProcessOverviewsByKey(
         OptimizeSearchRequestBuilderES.of(
             b ->
                 b.optimizeIndex(esClient, PROCESS_OVERVIEW_INDEX_NAME)
-                    .query(q -> q.ids(i -> i.values(processDefinitionKeys.stream().toList()))));
+                    .query(q -> q.ids(i -> i.values(processDefinitionKeys.stream().toList())))
```
Comment (grlimacan) [inline optimize/backend/src/main/java/io/camunda/optimize/service/db/repository/es/ProcessRepositoryES.java:62]: According to the ticket, the issue is also present with OpenSearch, but the fix is only for ElasticSearch?

**Review by grlimacan — CHANGES_REQUESTED** [review]
According to the ticket, the issue is also present with OpenSearch, but the fix is only for ElasticSearch?

File: optimize/backend/src/main/java/io/camunda/optimize/service/db/repository/es/ProcessRepositoryES.java (hunk 1 above)
Comment (oleksandr-kriuchenko-lohika) [reply optimize/backend/src/main/java/io/camunda/optimize/service/db/repository/es/ProcessRepositoryES.java:62]: No, for OpenSearch both code and UI behaiour look good.

**Comment by oleksandr-kriuchenko-lohika** [comment]
> LGTM, this should also be backported to stable/optimize-8.6 though . Is this error also present in C7?

This error should be present in C7. I was not able to reproduce it yesterday in C7, but now I can see that a prerequisite for this error is that number of process definitions should be more than 10. Anyway this fix is relevant for C7 as well I suppose. So yes, we'll have to backport it.

**Review by grlimacan — APPROVED** [review]
LGTM! I added a tag to backport to stable/optimize-8.6

**Pull request merged**